src/Umbraco.Web/Routing/PublishedContentRequestEngine.cs

```diff
@@ -630,20 +630,11 @@ private void FindTemplate()
 
 				var templateId = _pcr.PublishedContent.TemplateId;
 
-				if (templateId > 0)
-				{
-					ProfilingLogger.Logger.Debug<PublishedContentRequestEngine>("{0}Look for template id={1}", () => tracePrefix, () => templateId);
-					var template = ApplicationContext.Current.Services.FileService.GetTemplate(templateId);
-					if (template == null)
-						throw new InvalidOperationException("The template with Id " + templateId + " does not exist, the page cannot render");
-					_pcr.TemplateModel = template;

```

Comment:
The logic is moved to GetTemplate(), so that the equivalent line on 695 is set to _pcr.TemplateModel on  line 636.